### ISSUE TITLE
Question: is it possible to know the current zoom level – months vs. days?

### ISSUE THREAD

**Issue by hckr** [namespace-ee/react-calendar-timeline#655]
Is there an event which fires when the zoom level changes, e.g. from days to months and vice versa (I don't use any other zoom levels).

**Comment by Ilaiwi** [comment]
yes `onTimeChange`

**Comment by hckr** [comment]
AFAIR `onTimeChange` gives timestamps for the beginning and end of the visible time range. What I'd like to know is when this change happens:

![months-days-view](https://user-images.githubusercontent.com/1078650/65505738-35453980-deca-11e9-85a6-bbe2fc4f8275.gif)

**Comment by Ilaiwi** [comment]
@hckr you can diff them and figure out the unit so for example if it was less than a day then the unit is hour....

**Comment by hckr** [comment]
I don't exactly understand, what do you mean.

With the following callback:

```js
function onTimeChange(start, end, updateScrollCanvas) {
  console.log(
    "days: ",
    moment(start).diff(end, "days"),
    "months:",
    moment(start).diff(end, "months")
  );
  updateScrollCanvas(start, end);
}
```

I don't get reasonable results (because this shows days in the visible range, not the zoom level: if the columns are days or months):

![months-days-view-diff](https://user-images.githubusercontent.com/1078650/65508237-0b8f1100-ded0-11e9-9189-cab713ae0e79.gif)

**Comment by hckr** [comment]
I found out that a better event to use would be `onZoom`.

Ok, so the functionality I need is implemented here: https://github.com/namespace-ee/react-calendar-timeline/blob/c358570024fc0969a4bdbfe217dcd59491056cf3/src/lib/utility/calendar.js#L94

But in order to use it, first I have to import it in a hacky way:
```js
import { getMinUnit } from 'react-calendar-timeline/lib/lib/utility/calendar'
```

And then use another hack – a ref to the `Timeline` – in order to call it with proper arguments:
```js
onZoom = ({ timelineWidth, visibleTimeStart, visibleTimeEnd }) => {
    if (this.timelineRef.current) {
      const zoom = visibleTimeEnd - visibleTimeStart;
      const minUnit = getMinUnit(
        zoom,
        timelineWidth,
        this.timelineRef.current.props.timeSteps
      );
      console.log(minUnit);
    }
  };
```

But the result is, that on every zoom change, I get the current unit used in columns.
Wroking draft: https://codesandbox.io/s/timeline-demo-zoom-level-solution-3s8ey

---

It'd be great if there would be a less hacky way, maybe an event `onZoomLevelChange` which would fire when the value of `minUnit` changes? At least this is what I had in mind.

**Comment by Ilaiwi** [comment]
@hckr this has implementation details which I don't recommend... I am still not sure what are you trying to get and why the difference between visibleTimeStart and visibleTimeEnd didn't do the trick?

**Comment by hckr** [comment]
I would like to know if colums display months or days. I need to display different items for months and for days.

**Comment by Ilaiwi** [comment]
@hckr what do you mean by columns here? the primary header (red one) or the secondary header (gray one). the method I mentioned gets you the unit for the gray one

**Comment by hckr** [comment]
I meant the grey one, and the code I posted above does what I need, but is hacky.

**Comment by Ilaiwi** [comment]
@hckr try this
https://codesandbox.io/s/timeline-demo-zoom-level-question-8896o

**Comment by hckr** [comment]
It doesn't work correctly:

![screenshot](https://user-images.githubusercontent.com/1078650/67099014-7aa11200-f1bd-11e9-8354-226e679cebaf.png)

**Comment by Ilaiwi** [comment]
this is correct. you can see more than a month on the screen

**Comment by Ilaiwi** [comment]
@hckr yes I understand now. I guess there is no direct way to do that then. I would suggest adding a unit param to `onZoom` and `onTimeChange` with the unit. 
please feel free to open a PR to pass those params to the functions

**Comment by hckr** [comment]
Okay, I will implement this.
